numba/core/typing/builtins.py

```diff
@@ -1103,8 +1103,11 @@ class MinValInfer(AbstractTemplate):
     def generic(self, args, kws):
         assert not kws
         assert len(args) == 1
-        assert isinstance(args[0], (types.DType, types.NumberClass))
-        return signature(args[0].dtype, *args)
+        if isinstance(args[0], (types.DType, types.NumberClass)):
+            return signature(args[0].dtype, *args)
+        elif isinstance(args[0], (types.NPDatetime, types.NPTimedelta)):
+            return signature(args[0], *args)
```

Comment:
This doesn't seems to be needed. the datetime types seems to always be wrapped in a `DType`; e.g. `dtype(timedelta64[D])`.